code/modules/clothing/head/misc.dm

```diff
@@ -175,7 +175,25 @@
 	name = "fedora"
 	icon_state = "fedora"
 	item_state = "fedora"
-	desc = "A really cool hat if you're a mobster. A really lame hat if you're not."
+	desc = "It's NOT a trilby!"
+	action_button_name = "Tip Fedora"
+
+/obj/item/clothing/head/fedora/New()
+	if(prob(50))
+		name = "trilby"
+		desc = "It's NOT a fedora!"
+
+/obj/item/clothing/head/fedora/attack_self()
+	if(usr.canmove)

```

Comment:
Add the standard if(tip_cooldown > world.time) kinda stuff to prevent this from being an infinite wall of text please